src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double RESIZE = 1.5;
+    private Object[] array;
```

Comment:
We can skip casting in the methods using T here
```suggestion
    private T[] array;
```